### PR TITLE
Use proper error type and clean-up code

### PR DESCRIPTION
**Pull request by LukasReschke**
cc @icewind1991 please review.

### CODE REVIEW

File: lib/private/arrayparser.php [hunk 1]
```diff
@@ -119,7 +124,7 @@ function parseString($string) {
 	 * @param string $string
 	 * @return int
 	 */
-	function parseNum($string) {
+	public function parseNum($string) {
```
Comment (icewind1991) [inline lib/private/arrayparser.php]: private?

Comment (LukasReschke) [inline lib/private/arrayparser.php]: Indeed…

**Comment by LukasReschke** [comment]
@owncloud-bot Retest this please.

**Comment by icewind1991** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
Do we have unit tests on this class?
We still need this class? I cannot remember to see this being used ....

**Comment by LukasReschke** [comment]
> Do we have unit tests on this class?

Optimist :see_no_evil: 

> We still need this class? I cannot remember to see this being used ....

It is used for legacy purposes since we to my understanding never did an update repair step. – When an old user with a `mount.php` logs-in this one gets parsed as there is no `mount.json` yet.

**Comment by LukasReschke** [comment]
@icewind1991 How hard would an actual repair step be?

**Comment by DeepDiver1975** [comment]
> It is used for legacy purposes since we to my understanding never did an update repair step. – When an old user with a mount.php logs-in this one gets parsed as there is no mount.json yet.

well - this is more then one major release away - right?

Let's start some healing fire ....

**Comment by LukasReschke** [comment]
Sure. But since there was no repair step this still happens if an older user logs in. :see_no_evil: 

But then again the question is how easy a repairstep is possible. Then we could let it die in the future.

Alternatively we just drop it and add some error to the OwnCloud log if we find a mount.php?

**Comment by DeepDiver1975** [comment]
> Sure. But since there was no repair step this still happens if an older user logs in. :see_no_evil:
> 
> But then again the question is how easy a repairstep is possible. Then we could let it die in the future.
> 
> Alternatively we just drop it and add some error to the OwnCloud log if we find a mount.php?

on the other hand we will move mount.json to the database in 8.2 - @PVince81 right?

Maybe this is the right time then to do a proper cleanup

**Comment by DeepDiver1975** [comment]
so ... :+1:

**Pull request merged**